### ISSUE TITLE
defineCommand  loads lua scripts several times

### ISSUE THREAD

**Issue by akaNightmare** [luin/ioredis#778]
Hello, 
while executing some custom lua scripts several times in pipeline can see that ioredis:
1. runs `SCRIPT EXISTS` command with the multiple same sha1 hashes
2. runs `SCRIPT LOAD` for each of custom command (in the example below -> `.test()`) in pipeline 

### So questions are:
1. Is it expected behaviour?
2. Is it possible to run `SCRIPT EXISTS` command with uniq sha1 hashes?
3. Is it possible to run `SCRIPT LOAD` only once for same sha1 hashes?

### Small repro
```js
const Redis = require('ioredis');

const redis = new Redis('redis://127.0.0.1:6379');
redis.defineCommand('test', {
    lua: 'return {KEYS[1],KEYS[2],unpack(ARGV)}'
});

(async () => {
    const pipe = redis.pipeline();
    pipe.test(2, 'k1', 'k2', 'a1', 'a2');
    pipe.test(1, 'k1', 'a1');
    pipe.test(3, 'k1', 'k2', 'k3', 'a1', 'a2', 'a3');
    await pipe.exec();
    process.exit(0);
})();

```

### Redis Logs
```
  ioredis:redis status[127.0.0.1:6379]: [empty] -> connecting +0ms
  ioredis:redis queue command[127.0.0.1:6379]: 0 -> script([ 'exists', '86efd0ce7b213e7c68b64afe58b81b21feafeae8', '86efd0ce7b213e7c68b64afe58b81b21feafeae8', '86efd0ce7b213e7c68b64afe58b81b21feafeae8' ]) +4ms
  ioredis:redis status[127.0.0.1:6379]: connecting -> connect +8ms
  ioredis:redis write command[127.0.0.1:6379]: 0 -> info([]) +0ms
  ioredis:redis status[127.0.0.1:6379]: connect -> ready +3ms
  ioredis:connection send 1 commands in offline queue +0ms
  ioredis:redis write command[127.0.0.1:6379]: 0 -> script([ 'exists', '86efd0ce7b213e7c68b64afe58b81b21feafeae8', '86efd0ce7b213e7c68b64afe58b81b21feafeae8', '86efd0ce7b213e7c68b64afe58b81b21feafeae8' ]) +1ms
  ioredis:redis write command[127.0.0.1:6379]: 0 -> script([ 'load', 'return {KEYS[1],KEYS[2],unpack(ARGV)}' ]) +0ms
  ioredis:redis write command[127.0.0.1:6379]: 0 -> script([ 'load', 'return {KEYS[1],KEYS[2],unpack(ARGV)}' ]) +1ms
  ioredis:redis write command[127.0.0.1:6379]: 0 -> script([ 'load', 'return {KEYS[1],KEYS[2],unpack(ARGV)}' ]) +0ms
  ioredis:redis write command[127.0.0.1:6379]: 0 -> evalsha([ '86efd0ce7b213e7c68b64afe58b81b21feafeae8', '2', 'k1', 'k2', 'a1', 'a2' ]) +0ms
  ioredis:redis write command[127.0.0.1:6379]: 0 -> evalsha([ '86efd0ce7b213e7c68b64afe58b81b21feafeae8', '1', 'k1', 'a1' ]) +1ms
  ioredis:redis write command[127.0.0.1:6379]: 0 -> evalsha([ '86efd0ce7b213e7c68b64afe58b81b21feafeae8', '3', 'k1', 'k2', 'k3', 'a1', 'a2', 'a3' ]) +0ms
```

Thanks.

**Comment by akaNightmare** [comment]
@luin going to create a PR

**Comment by luin** [comment]
Released with v4.5.1
